shared/events/edgeware/filters/enricher.ts

```diff
@@ -496,12 +496,17 @@ export default async function (
        */
       case SubstrateEventKind.Offence: {
         const [ offenceKind, opaqueTimeSlot, applied ] = event.data as unknown as [ Kind, OpaqueTimeSlot, boolean ];
+        const reportIds = await api.query.offences.concurrentReportsIndex(offenceKind, opaqueTimeSlot);
+        const offenceDetails: Option<OffenceDetails>[] = await api.query.offences.reports
+          .multi(reportIds.map((reportId) => reportId.toString()));
+
         return {
           data: {
             kind,
             offenceKind: offenceKind.toString(),
             opaqueTimeSlot: opaqueTimeSlot.toString(),
-            applied: typeof applied === 'undefined' ? true : applied
+            applied: typeof applied === 'undefined' ? true : applied,
+            offenceDetails
```

Comment:
Ideally we could avoid storing Polkadot types directly on these objects -- we need them to be JSON-ifiable because they end up in the DB.

Also, we should filter out the None options and unwrap the others, to avoid having gaps in the array.